tests/rptest/tests/datalake/3rdparty_rewrite.py

```diff
@@ -0,0 +1,118 @@
+        self.table_name = f"redpanda.{self.topic_name}"
+
+    def setUp(self):
+        # redpanda will be started by DatalakeServices
+        pass
+
+    @cluster(num_nodes=4)
+    @matrix(cloud_storage_type=supported_storage_types(),
+            query_engine=[QueryEngineType.SPARK, QueryEngineType.TRINO],
+            filesystem_catalog_mode=[True, False])
+    def test_e2e_basic(self, cloud_storage_type, query_engine,
+                       filesystem_catalog_mode):
+        """
+        This test verifies that Redpanda can continue to work with Iceberg
+        metadata written by third-party query engines. We use an optimize operation
+        with a third-party query engine to trigger a rewrite of the data files
+        and metadata.
+        """
+        num_messages = 100
+        num_kafka_partitions = 10
+
+        with DatalakeServices(self.test_ctx,
+                              redpanda=self.redpanda,
+                              filesystem_catalog_mode=filesystem_catalog_mode,
+                              include_query_engines=[query_engine]) as dl:
+            dl.create_iceberg_enabled_topic(self.topic_name,
+                                            partitions=num_kafka_partitions)
+            dl.produce_to_topic(self.topic_name, 1024, num_messages)
+            dl.wait_for_translation(self.topic_name, msg_count=num_messages)
+
+            if query_engine == QueryEngineType.TRINO:
+                trino = dl.trino()
+
+                # See Trino metadata tables documentation
+                # https://trino.io/docs/current/connector/iceberg.html#alter-table-execute
+                initial_parquet_files = trino.count_table(
+                    "redpanda", f"{self.topic_name}$files")
+
+                assert initial_parquet_files >= num_kafka_partitions, f"Expecting at least {num_kafka_partitions} files, got {initial_parquet_files}"
+
+                # Optimize the table to rewrite the data.
+                # https://trino.io/docs/current/connector/iceberg.html#alter-table-execute
+                trino.run_query_fetch_all(
+                    f"ALTER TABLE redpanda.{self.topic_name} EXECUTE optimize")
+
+                optimized_parquet_files = trino.count_table(
+                    "redpanda", f"{self.topic_name}$files")
+                assert optimized_parquet_files == 1, f"Expecting 1 file after optimize, got {optimized_parquet_files}"
+            elif query_engine == QueryEngineType.SPARK:
+                spark = dl.spark()
+
+                initial_parquet_files = spark.run_query_fetch_all(
+                    f"SELECT count(*) FROM redpanda.{self.topic_name}.files"
+                )[0][0]
+                assert initial_parquet_files >= num_kafka_partitions, f"Expecting at least {num_kafka_partitions} files, got {initial_parquet_files}"
+
+                spark.run_query_fetch_all(
+                    f"CALL `redpanda-iceberg-catalog`.system.rewrite_data_files(\"redpanda.{self.topic_name}\")"
```

Comment:
This line here is what actually triggers compaction (along with the trino command above).

We should have a `datalake_services::do_maintenance_work()` function for future re-use, which triggers these spark/trino queries.

Reply:
I'd rather not generalize it until I have some experience with some additional vendors. Not sure what the method should be called and what work/how much work it should do yet. Extracted it to a separate method in the test class for now.

Reply:
sounds good!

Reply:
Actually, came up with a generalization. Also added a method for checking which engines support maintenance jobs. We should be able to build a grid with ticks re what is tested later 👌